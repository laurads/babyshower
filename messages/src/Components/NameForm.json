[
  {
    "id": "NameForm.name-placeholder",
    "defaultMessage": "Ton nom"
  },
  {
    "id": "NameForm.hiTitle",
    "defaultMessage": "Salut {name}"
  },
  {
    "id": "NameForm.hiDescription",
    "defaultMessage": "Nous avons besoin de ton aide pour trouver un prénom pour notre baby girl, et en prime tu peux jouer pour gagner une bouteille de champagne"
  },
  {
    "id": "NameForm.question",
    "defaultMessage": "Tape ton nom/prénom en entier ou ton surnom pour pouvoir accéder aux jeux"
  },
  {
    "id": "NameForm.validate",
    "defaultMessage": "Valider"
  }
]