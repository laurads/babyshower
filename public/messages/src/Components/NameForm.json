[
  {
    "id": "NameForm.validate",
    "defaultMessage": "Valider"
  },
  {
    "id": "NameForm.name-placeholder",
    "defaultMessage": "Ton nom"
  },
  {
    "id": "NameForm.question",
    "defaultMessage": "Rentre ton nom pour pouvoir acc√©der aux jeux"
  }
]