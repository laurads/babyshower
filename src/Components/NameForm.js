import React, { Component } from 'react';
import './ComponentStyle.css';
import { Button, Input, Form} from 'semantic-ui-react';
import {FormattedMessage, defineMessages, injectIntl} from 'react-intl';

const messages = defineMessages({
    namePlaceholder: {
        id: "NameForm.name-placeholder",
        defaultMessage: "Ton nom",
      },
});

class NameForm extends Component {
    constructor(props) {
    super(props);
        this.state = {
            name: '',
            nameValidated: false
        };
    }

    handleChange = (event) =>{
        const target = event.target;
        if(target){
            this.setState({
            [ target.name]: target.value
            });
        }
    }

    storeName = (event) => {
        event.preventDefault();
        this.setState({
            nameValidated: true
        })
        this.props.updatePlayerName(this.state.name);
    }

    render() {
        const {intl} = this.props;
        const {name} = this.state;
        const placeholder = intl.formatMessage(messages.namePlaceholder);
        return (
            <div >
                {!this.state.nameValidated && 
                    <Form onSubmit={this.storeName} className="Component-form">
                        <label className="Form-label">
                            <FormattedMessage
                                id="NameForm.question"
                                defaultMessage="Rentre ton nom pour pouvoir accéder aux jeux"
                            />
                        </label>
                        <div className="Form-row">
                            <Input 
                                className="Form-input"
                                type="text" 
                                name="name"
                                placeholder={placeholder}
                                value={this.state.name} onChange={this.handleChange}
                                required/>                    
                            <Button 
                            className="Form-button"> 
                                <FormattedMessage
                                    id="NameForm.validate"
                                    defaultMessage="Valider"
                                />
                            </Button>
                        </div>
                    </Form>
                }
                {this.state.nameValidated &&
                    <div className="Welcome-label">
                        <div className="Welcome-title">
                            <FormattedMessage
                                id="NameForm.hiTitle"
                                defaultMessage={'Salut {name}'}
                                values={{name:name }}
                            />
                        </div>
                        <div className="Welcome-description">  
                            <FormattedMessage
                                id="NameForm.hiDescription"
                                defaultMessage="Nous avons besoin de ton aide pour trouver un prénom pour notre baby girl, et en prime tu peux jouer pour gagner une bouteille de champagne"
                            />
                        </div>
                    </div>
                }
            </div>
        );
    }
}

export default injectIntl(NameForm);