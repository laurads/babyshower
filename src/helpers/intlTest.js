import React from 'react';
import { IntlProvider, intlShape } from 'react-intl';
import { mount, shallow } from 'enzyme';

const messages = require('../../public/locales/en'); // en.json
const intlProvider = new IntlProvider({ locale: 'en', messages }, {});
const { intl } = intlProvider.getChildContext();

/**
 * When using React-Intl `injectIntl` on components, props.intl is required.
 */
function nodeWithIntlProp(node) {
  return React.cloneElement(node, { intl });
}

export function shallowWithIntl(node) {
    return shallow(nodeWithIntlProp(node), { 
      context: { intl } 
    });
  }

export function mountWithIntl(node) {
  return mount(nodeWithIntlProp(node), {
    context: { intl },
    childContextTypes: { intl: intlShape }
  });
}