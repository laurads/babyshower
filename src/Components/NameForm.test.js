import React from 'react';
import NameForm from './NameForm';
import Enzyme, {shallow, mount} from 'enzyme';
import {Form, Input, Button} from 'semantic-ui-react';
import jsdom from 'jsdom';
import 'jsdom-global/register';
import {mountWithIntl, shallowWithIntl} from '../helpers/intlTest';

describe('<NameForm />', () => {
    const wrapper = shallowWithIntl(<NameForm/>);

    it('should render correctly with no props', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a Form by default', () => {
        const form = wrapper.find('Form').first();
        expect(form).toBeDefined();
    });

    it('should render an Input by default', () => {
        const input = wrapper.find('Input').first();
        expect(input).toBeDefined();
    });

    it('should render a Button by default', () => {
        const button = wrapper.find('Button').first();
        expect(button).toBeDefined();
    });

    /*it('should call handleChange function when fill the input in', () => {
        const updatePlayerName = jest.fn();
        const wrapperIntl = mountWithIntl(<NameForm updatePlayerName={updatePlayerName}/>);
        wrapperIntl.instance().handleChange = jest.fn();
        wrapperIntl.instance().forceUpdate();
        wrapperIntl.update();
        console.log(wrapperIntl.find('Input').debug());
        wrapperIntl.find('Input').simulate('change', { target : {value: 'newName' }});
        expect(wrapperIntl.instance().handleChange).toHaveBeenCalled();
        wrapperIntl.unmount();
    });

    it('should update state name when fill the input in', () => {
        const wrapperIntl = mountWithIntl(<NameForm/>);
        const newValue = 'newName';
        wrapperIntl.find('input').simulate('change', { target : {value: newValue }});
        const name = wrapperIntl.state().name;
        expect(name).toEqual(newValue);
        wrapperIntl.unmount();
    });

    it('should call handleValidateSubmit function on submit', () => {
        const updatePlayerName = jest.fn();
        const wrapperIntl = mountWithIntl(<NameForm updatePlayerName={updatePlayerName}/>);
        wrapperIntl.instance().storeName = jest.fn();
        wrapperIntl.instance().forceUpdate();
        wrapperIntl.update();
        wrapperIntl.find('Form').simulate('submit');
        expect(wrapperIntl.instance().storeName).toHaveBeenCalled();
        wrapperIntl.unmount();
    });*/

    it('should call validateForm function on submit', () => {
        const updatePlayerName = jest.fn();
        const wrapperIntl = mountWithIntl(<NameForm updatePlayerName={updatePlayerName}/>);
        wrapperIntl.find('Form').simulate('submit');
        expect(updatePlayerName).toHaveBeenCalled();
        wrapperIntl.unmount();
    });
});