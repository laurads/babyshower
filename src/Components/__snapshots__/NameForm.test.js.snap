// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NameForm /> should render correctly with no props 1`] = `ShallowWrapper {}`;
